odl/discr/partition.py

```diff
@@ -451,7 +451,14 @@ def __eq__(self, other):
             return True
 
         # Optimized version for exact equality
-        return self.set == other.set and self.grid == other.grid
+        return (isinstance(other, type(self)) and
```

Comment:
Main reason was that it was flagged by Pylint as a code smell - and unidiomatic way of doing things. The point is not that it's unidiomatic regarding what we want to achieve (checking equivalence of types) but rather that the fact of enforcing an exact type is deemed unidiomatic by the linter.
However, since we need exactly that here, we can as well go with the simpler code and ignore the flag.